docs/develop-an-agent.md

```diff
@@ -0,0 +1,423 @@
+        https://github.com/Azure/iot-hub-device-update/blob/7a6e4c12d54df7a92984e935c3050f35e359a342/src/agent/src/main.c#L656 
+
+   1. Using the OpenSSL engine with the C SDK
+
+        https://github.com/Azure/iot-hub-device-update/blob/7a6e4c12d54df7a92984e935c3050f35e359a342/src/agent/src/main.c#L518
+
+---
+
+This document covers how a Linux process (e.g. a systemd service) can leverage APIs exposed by the [identity](../identity-service.md) (IS), [keys](../keys-service.md) (KS), and [certificates](../certificates-service.md) (CS) services to establish a connection to Azure IoT Hub using a [module identity](https://docs.microsoft.com/en-us/azure/iot-hub/iot-hub-devguide-module-twins). A _module identity_ is the same cloud identity used by [Edge modules](https://docs.microsoft.com/azure/iot-edge/iot-edge-modules#module-identities) to communicate with IoT Hub. 
+
+The APIs for IS/KS/CS are all exposed to processes on the local host over Unix Domain Sockets (UDS). The services enable your agent (or, more generally, any authorized process) to co-exist side by side with other agents as well as IoT Edge modules (if IoT Edge is installed). They all appear as modules under the same logical device in IoT Hub and can be accessed via your back-end cloud solution. 
+
+
+> **Note**
+>
+> The IS/KS/CS services can also be used by a process to connect to Hub using the device identity. However, in practice, that should not be used. The device identity is more privileged than a module identity as it has ownership for the module identities. 
+>
+> Furthermore, two separate processes cannot simultaneously connect to Hub using the same identity. Unless you own the device and can guarantee that only your process will ever use the device identity, you should stick to using a module identity. 
+
+**Differences with Edge modules**
+
+[IoT Central](https://docs.microsoft.com/azure/iot-central/core/overview-iot-central) only supports Edge modules. It does not provide UI for arbitrary agents using a module identity to connect to Azure.
+
+The Identity Service package does not provide non-Edge modules with higher-level capabilities that come with IoT Edge.
+
+Edge modules are managed through the [IoT Edge runtime](https://docs.microsoft.com/azure/iot-edge/about-iot-edge#iot-edge-runtime) which enables:
+- Deploying cloud workloads to an edge device
+- Monitoring the health of the deployed Edge modules
+- Gathering and reporting runtime quality metrics
+- Managing communication between Edge modules
+- Remotely restarting a module or retrieving logs
+
+## Prerequisites
+
+### Installation
+
+The IS/KS/CS services are installed as part of the [aziot-identity-service package](../packaging.md). 
+
+> **Note**
+>
+> By design the package conflicts with the `iotedge` and `libiothsm-std` packages. If you're using `apt` to install the package it will prompt to remove the conflicting packages.  Otherwise, be sure to manually remove them before installing the Identity Service.
```

Comment:
Mention that `iotedge` and `libiothsm-std` are the 1.1 packages. So people know that installing `aziot-identity-service` is a version upgrade (and not backwards compatible).